tests/unit_tests/test_optional-enhancements.py

```diff
@@ -0,0 +1,347 @@
+import pytest
+from swap_meet.vendor import Vendor
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+from swap_meet.item import Item
+
+# @pytest.mark.skip
+def test_get_newest_by_category():
+    # Arrange
+    item_a = Clothing(age=2)
+    item_b = Decor(age=2)
+    item_c = Clothing(age=4)
+    item_d = Decor(age=5)
+    item_e = Clothing(age=3)
+    sam = Vendor(
+        inventory=[item_a, item_b, item_c, item_d, item_e]
+    )
```

Comment:
Love seeing this and all the other tests you added for the optional functionality!!!